services:
  certbot:
    image: certbot/certbot
    restart: no
    networks:
      - certbot
    volumes:
      - hive_certbot_certs:/etc/letsencrypt
      - certbot_webroot:/var/www/letsencrypt
      - certbot_workdir:/var/lib/letsencrypt
    # This default command means it'll print the message and exit with
    # status 0 on "docker compose up", the "restart: no" above stops
    # compose worrying about this, and anyone trying to debug this will
    # presumably run "docker compose logs certbot" and see the message.
    command:
      - certonly
      - --help
    profiles:
      - certbot

  certbot-autorenew:
    image: certbot/certbot
    init: true
    restart: unless-stopped
    networks:
      - certbot
    volumes:
      - hive_certbot_certs:/etc/letsencrypt
      - certbot_webroot:/var/www/letsencrypt
      - certbot_workdir:/var/lib/letsencrypt
      - ./hive/certbot:/opt/hive:ro
      - hive_certdist_root:/var/lib/hive/certdist
    entrypoint: python
    command: /opt/hive/autorenew.py
    profiles:
      - ingress

  nginx-http:
    image: nginx
    restart: unless-stopped
    networks:
      - nginx-http
    ports:
      - 80:80/tcp
    volumes:
      - ./default.conf:/etc/nginx/conf.d/default.conf:ro
      - certbot_webroot:/var/www/letsencrypt:ro
    profiles:
      - ingress

networks:
  certbot:
  nginx-http:

volumes:
  certbot_webroot:
  certbot_workdir:
  hive_certbot_certs:
    external: true
  hive_certdist_root:
    external: true
  hive_certdist_host:
    external: true
